### PR TITLE
Get connectivity rpc

### PR DESCRIPTION
**Pull request by asahtik**

### CODE REVIEW

File: include/depthai/device/DeviceBase.hpp [hunk 1]
```diff
@@ -570,6 +571,13 @@ class DeviceBase {
      */
     std::vector<CameraBoardSocket> getConnectedCameras();
 
+    /**
+     * Get connectivity for device
+     *
+     * @returns Vector of connection type
+     */
+    std::vector<Connectivity> getConnectivity();
```
Comment (moratom) [inline include/depthai/device/DeviceBase.hpp:579]: I think something like `getAvailableInterfaces()` would be better here. 
And `[Connection?]Interface` for the type.

**Review by moratom — APPROVED** [review]
Apart from the naming, LGTM!

File: include/depthai/device/DeviceBase.hpp (hunk 1 above)
Comment (themarpe) [reply include/depthai/device/DeviceBase.hpp]: Note Available usually denotes a subset of all existing. In this case, the function should return "all" device's interfaces that it has. (the "current / what is connectable by" is explorable through device discovery)

Lets just document it accordingly and LGTM

Comment (moratom) [reply include/depthai/device/DeviceBase.hpp]: @themarpe just to clarify, do you mean leave it at `getConnectivity()` or changing it to `getAllInterfaces()` ?

Comment (themarpe) [reply include/depthai/device/DeviceBase.hpp]: Perhaps `getConnectionInterfaces` and later on / if need be we can expose available interfaces

**Pull request merged**